setup/Setup.Core/PatchTask.cs

```diff
@@ -0,0 +1,198 @@
+			? [
+				..(await File.ReadAllLinesAsync(removedFileList, cancellationToken).ConfigureAwait(false)).Select(PathUtils.SetCrossPlatformDirectorySeparators),
+			]
+			: [];
+
+		var items = new List<WorkItem>();
+		var newFiles = new ConcurrentDictionary<string, object?>();
+
+		foreach ((string file, string relPath) in EnumerateFiles(parameters.PatchDir)) {
+			if (relPath.EndsWith(".patch")) {
+				items.Add(new WorkItem("Patching: " + relPath, async ct => {
+					FilePatcher filePatcher = await Patch(file, ct).ConfigureAwait(false);
+					newFiles.TryAdd(PathUtils.SetCrossPlatformDirectorySeparators(filePatcher.PatchedPath), null);
+				}));
+				noCopy.Add(relPath.Substring(0, relPath.Length - 6));
+			}
+			else if (relPath != DiffTask.RemovedFileList) {
+				string destination = PathUtils.SetCrossPlatformDirectorySeparators(Path.Combine(parameters.PatchedDir, relPath));
+
+				items.Add(new WorkItem("Copying: " + relPath, () => Copy(file, destination)));
+				newFiles.TryAdd(destination, null);
+			}
+		}
+
+		foreach ((string file, string relPath) in EnumerateSrcFiles(parameters.BaseDir)) {
+			if (!noCopy.Contains(relPath)) {
+				string destination = PathUtils.SetCrossPlatformDirectorySeparators(Path.Combine(parameters.PatchedDir, relPath));
+
+				items.Add(new WorkItem("Copying: " + relPath, () => Copy(file, destination)));
+				newFiles.TryAdd(destination, null);
+			}
+		}
+
+		try {
+			CreateDirectory(programSettings.LogsDir);
+			logFile = new StreamWriter(Path.Combine(programSettings.LogsDir, "patch.log"));
+
+			progress.Report(new ProgressInfo { MaxProgress = items.Count });
+			await ExecuteParallel(items, progress, cancellationToken: cancellationToken).ConfigureAwait(false);
+		}
+		finally {
+			logFile?.Close();
+		}
+
+		parameters.Cutoff.Set(DateTime.Now);
+
+		//Remove files and directories that weren't in patches and original src.
+
+		progress.Report(new ProgressInfo { Status = "Deleting Old Src Files" });
+
+		foreach ((string file, _) in EnumerateSrcFiles(parameters.PatchedDir)) {
+			if (!newFiles.ContainsKey(file)) {
+				File.Delete(file);
+			}
+		}
+
+		progress.Report(new ProgressInfo { Status = "Deleting Old Src's Empty Directories" });
+
+		DeleteEmptyDirs(parameters.PatchedDir);
+
+		progress.Report(new ProgressInfo { Status = "Old Src Removed" });
+
+		//Show patch reviewer if there were any fuzzy patches.
+
+		if (fuzzy > 0 || (programSettings.PatchMode == Patcher.Mode.FUZZY && failures > 0)) {
+			this.patchReviewer?.Show(results, commonBasePath: parameters.BaseDir + '/');
+		}
+	}
+
+	public override bool Failed() => failures > 0;
+	public override bool Warnings() => warnings > 0;
+
+	public override void FinishedPrompt()
+	{
+		if (fuzzy > 0) {
+			return;
+		}
+
+		this.userPrompt.Prompt(
+			"Patch Results",
+			$"Patches applied with {failures} failures and {warnings} warnings.\nSee /logs/patch.log for details",
+			Failed() ? PromptSeverity.Error : PromptSeverity.Warning);
```

Comment:
Perhaps this should be called `Inform` rather than `Prompt`? Just so it's clear that it still runs even when `NoPrompts` is set

Reply:
Renamed. I struggled a bit with the naming there.